.github/workflows/autopilot-branch-to-latest.yaml

```diff
@@ -0,0 +1,254 @@
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+  TOOLSUITE_CIDR: 10.1.0.0/16
+
+jobs:
+  # build-tool is only responsible for building the 'tool' and making it available
+  # to other jobs via a unique GHA cache.
+  build-tool:
+    name: Build Tool
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+
+    - name: Build Tool
+      run: |
+        cd hack/tool
+        make
+
+    - name: Save Build Tool Image
+      run: |
+        docker save tool:latest | gzip > tool-latest.tar.gz
+
+    - name: Cache Tool Image
+      uses: actions/cache@v3
+      with:
+        key: tool-${{ github.sha }}
+        path: tool-latest.tar.gz
+
+  # setup provisions the VPC infrastructure and caches terraform state, along with preserving the
+  # generated VPC id.
+  setup:
+    name: Setup Infrastructure
+    runs-on: ubuntu-latest
+    needs: [ build-tool ]
+
+    outputs:
+      vpcid: ${{ steps.vpcid.outputs.vpcid }}
+
+    steps:
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v2
+      with:
+        terraform_wrapper: false
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+    - name: Cache Tool Image (restore)
+      uses: actions/cache@v3
+      with:
+        key: tool-${{ github.sha }}
+        path: tool-latest.tar.gz
+
+    - name: Import Tool
+      run: |
+        docker load --input tool-latest.tar.gz
+        rm tool-latest.tar.gz
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+
+    - name: Create VPC
+      run: |
+        cd hack/tool
+        export TOOL_DATADIR=$(pwd)/infra
+        mkdir -p ${TOOL_DATADIR}
+        ./tool.sh aws vpcinfra create \
+          --name k0sts-infra \
+          --region ${{ github.event.inputs.region }} \
+          --cidr ${{ env.TOOLSUITE_CIDR }}
+
+    - name: Save VPC ID
+      id: vpcid
+      run: |
+        cd hack/tool/infra
+        echo "vpcid=$(terraform output -raw vpc_id)" >> $GITHUB_OUTPUT
+
+    - name: Cache Infrastructure Terraform
+      uses: actions/cache@v3
+      with:
+        key: terraform-${{ github.sha }}
+        path: hack/tool/infra/terraform.tfstate
+
+  # test runs through a matrix of from/to k0s version upgrades. This leverages the 'tool'
+  # to create k0s clusters using k0sctl, and automatically generates autopilot Plans to
+  # perform updates to all nodes.
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [ setup ]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # The 'testdata' matrix focuses on grouping updates of individual historic branch releases
+        # to the latest released version on a branch.
+        testdata: [
+          # v1.24 versions
+          {
```

Comment:
I wonder if we want to have this matrix auto-generated as in the "Go build" workflow?

Reply:
That's an interesting thought. We still need some authority to answer 'what are the supported autopilot upgrade paths'.  Whether it's done here or somewhere else really makes no difference to me. I'm open to ideas on this.

Reply:
#2345